mitosheet/src/components/taskpanes/SplitTextToColumns/SplitTextToColumnsTaskpane.tsx

```diff
@@ -0,0 +1,302 @@
+
+        if (params !== undefined && params.column_id !== undefined) {
+            setLoadingPreview(true);
+
+            const _splitTextToColumnsPreviewArray = await props.mitoAPI.getSplitTextToColumnsPreview(params.sheet_index, params.column_id, params.delimiters)
+            if (_splitTextToColumnsPreviewArray !== undefined) {
+                setPreview(_splitTextToColumnsPreviewArray)
+            } else {
+                setPreview(undefined)
+            }
+
+            setLoadingPreview(false);
+        } else {
+            setPreview(undefined)
+        }
+    }
+
+    // Keep track of the delimiters in this object so we don't need to messily convert back and forth between 
+    // delimiter name and character. 
+    const [delimiterObjs, setDelimiterObjs] = useState<Record<string, DelimiterObj>>(defaultDelimitersObj)
+
+    // When the delimitersObj changes, update the delimiter param
+    useDebouncedEffect(() => {
+        setParams(prevParams => {
+            return {
+                ...prevParams,
+                delimiters: getIncludedDelimiters()
+            }
+        })
+    }, [delimiterObjs], 500)
+
+    useEffect(() => {
+        if (params !== undefined && params.delimiters.length > 0) {
+            // If there is at least one delimiter, load the preview
+            void loadSplitTextToColumnsPreview()
+        } else {
+            setPreview(undefined)
+        }
+    }, [params])
+
+    // Returns a list of delimiter characters from the delimiter object
+    const getIncludedDelimiters = (): string[] => {
+        return Object.values(delimiterObjs).filter(delimiterObj => delimiterObj.included).map((delimiterObj) => delimiterObj.delimiter)
+    }
+    
+    if (params === undefined || params.column_id === undefined) {
+        return (<DefaultEmptyTaskpane setUIState={props.setUIState}/>)
+    }
+
+    return (
+        <DefaultTaskpane>
+            <DefaultTaskpaneHeader 
+                header="Split Text to Column"
+                setUIState={props.setUIState}            
+            />
+            <DefaultTaskpaneBody>
+                <Row justify='space-between' align='center'>
+                    <Col>
+                        <p className='text-header-3'>
+                            Sheet
+                        </p>
+                    </Col>
+                    <Col>
+                        <Select
+                            width='medium'
+                            value={props.dfNames[params.sheet_index]}
+                            // Safe to cast as dfNames are strings
+                            onChange={(newSheet: string) => {
+                                setParams(prevParams => {
+                                    const newSheetIndex = props.dfNames.indexOf(newSheet)
+                                    return {
+                                        ...prevParams,
+                                        sheet_index: newSheetIndex
```

Comment:
I think we should change the column id here, rather than doing all this work at the top. Just update to the default selected column id in the new sheet (which is the first?).